src/common/lib/util/hashring.ts

```diff
@@ -0,0 +1,44 @@
+// HashRing currently implements a modulo-based hash partitioning scheme.
+// It is used to distribute keys across a set of nodes.
+// TODO(mschristensen): use consistent hashing instead of modulo-based hashing
+// to avoid re-assigning all keys when nodes are added or removed.
+export default class HashRing {
```

Comment:
Small suggestion: maybe it's better to rename class? I believe name HashRing implies consistent hashing, but until you look at implementation or read TODO, you never know it's module-based hash